runtime/src/main/kotlin/kotlin/native/concurrent/SharedRef.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package kotlin.native.concurrent
+
+import kotlin.native.internal.*
+
+@SymbolName("Kotlin_SharedRef_createSharedRef")
+external private fun createSharedRef(value: Any): NativePtr
+
+/**
+ * A frozen shared reference to a Kotlin object
+ *
+ * Can be safely passed between workers, but can only be accessed on the worker it was created on
```

Comment:
Unless the referred object is frozen too.

Reply:
Curious, what's the purpose/use-case for a `SharedRef`?

Reply:
The purpose is to share reference without sharing contents. In particular, to have a reference from shared object to non-shared object. For example, when some methods of the shared object can be used from any execution context, while others must be called only from the particular worker and rely on objects of this worker.